### ISSUE TITLE
Parsing error when using ternary operator

### ISSUE THREAD

**Issue by dc42** [dc42/RepRapFirmware#1021]
Forum thread https://forum.duet3d.com/topic/36017/unexpected-error-with-and-ternary-operator.

Given this macro:
```
var A = {{1,2}, {3,4}}
echo (true ? 42 : #var.A[0])
echo (true ? 42 : #(var.A[0]))
```
the first echo command outputs 42 as expected but the second one provokes this error message:
`Error: in file macro line 3 column 30: meta command: expected object model value or string after '#`
If 'true' is replaced in that line by 'false' then 2 is output as expected.

**Comment by dc42** [comment]
See also issue 1020 for a separate issue related to this example

**Comment by dc42** [comment]
Fixed and test in 3.6-dev and back-ported to 3.5-dev.
